cmd/lncli/walletrpc_active.go

```diff
@@ -168,3 +173,130 @@ func bumpFee(ctx *cli.Context) error {
 
 	return nil
 }
+
+var bumpChannelCloseCommand = cli.Command{
+	Name:      "bumpchannelclose",
+	Usage:     "Bumps the fee of a channel closing transaction.",
+	ArgsUsage: "channel_point",
+	Description: `
+	This command allows the fee of a channel closing transaction to be
+	increased by using the child-pays-for-parent mechanism. It will instruct
+	the sweeper to sweep the anchor outputs of transactions in the set
+	of valid commitments for the specified channel at the requested fee
+	rate or confirmation target.
+	`,
+	Flags: []cli.Flag{
+		cli.Uint64Flag{
+			Name: "conf_target",
+			Usage: "the number of blocks that the output should " +
+				"be swept on-chain within",
+		},
+		cli.Uint64Flag{
+			Name: "sat_per_byte",
+			Usage: "a manual fee expressed in sat/byte that " +
+				"should be used when sweeping the output",
+		},
+	},
+	Action: actionDecorator(bumpChannelClose),
+}
+
+func bumpChannelClose(ctx *cli.Context) error {
+	// Display the command's help message if we do not have the expected
+	// number of arguments/flags.
+	if ctx.NArg() != 1 {
+		return cli.ShowCommandHelp(ctx, "bumpchannelclose")
+	}
+
+	// Validate the channel point.
+	channelPoint := ctx.Args().Get(0)
+	_, err := NewProtoOutPoint(channelPoint)
+	if err != nil {
+		return err
+	}
+
+	// Fetch all waiting close channels.
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	// Fetch waiting close channel commitments.
+	commitments, err := getWaitingCloseCommitments(client, channelPoint)
+	if err != nil {
+		return err
+	}
+
+	// Retrieve pending sweeps.
+	walletClient, cleanUp := getWalletClient(ctx)
+	defer cleanUp()
+
+	ctxb := context.Background()
+	sweeps, err := walletClient.PendingSweeps(
+		ctxb, &walletrpc.PendingSweepsRequest{},
+	)
+	if err != nil {
+		return err
+	}
+
+	// Match pending sweeps with commitments of the channel for which a bump
+	// is requested and bump their fees.
+	commitSet := map[string]struct{}{
+		commitments.LocalCommitHash:  {},
+		commitments.RemoteCommitHash: {},
+	}
+	if commitments.RemotePendingCommitHash != "" {
+		commitSet[commitments.RemotePendingCommitHash] = struct{}{}
+	}
+
+	for _, sweep := range sweeps.PendingSweeps {
+		// Only bump anchor sweeps.
+		if sweep.WitnessType != walletrpc.WitnessType_COMMITMENT_ANCHOR {
+			continue
+		}
+
+		// Skip unrelated sweeps.
+		sweepTxID, err := chainhash.NewHash(sweep.Outpoint.TxidBytes)
+		if err != nil {
+			return err
+		}
+		if _, match := commitSet[sweepTxID.String()]; !match {
+			continue
+		}
+
+		// Bump fee of the anchor sweep.
+		fmt.Printf("Bumping fee of %v:%v\n",
+			sweepTxID, sweep.Outpoint.OutputIndex)
+
+		_, err = walletClient.BumpFee(ctxb, &walletrpc.BumpFeeRequest{
+			Outpoint:   sweep.Outpoint,
+			TargetConf: uint32(ctx.Uint64("conf_target")),
+			SatPerByte: uint32(ctx.Uint64("sat_per_byte")),
+			Force:      true,
+		})
```

Comment:
Problem to solve: this call clears the exclusive group.

Reply:
Fixed by making exclusive group static, see commit "sweep: leave exclusive group unchanged on parameter update"